pkg/cgroup/cgroup.go

```diff
@@ -0,0 +1,312 @@
+package cgroup
+
+import (
+	"bufio"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/aquasecurity/tracee/pkg/mount"
+)
+
+//
+// Constants
+//
+
+const procCgroups = "/proc/cgroups"
+const procFilesystems = "/proc/filesystems"
+
+const CgroupV1FsType = "cgroup"
+const CgroupV2FsType = "cgroup2"
+
+const DefaultCgroupController = "cpuset"
+
+const TmpCgroup = "/tmp"
+const TmpCgroupV1Path = "tracee-cgroupv1"
+const TmpCgroupV2Path = "tracee-cgroupv2"
+
+type CgroupVersion int
+
+const (
+	CgroupV1 CgroupVersion = iota + 1
+	CgroupV2
+)
+
+//
+// Cgroups
+//
+
+// Cgroups handles cgroups initialization (v1+v2 or v2 only)
+type Cgroups struct {
+	cgroupv1 *Cgroup // used for cgroup filesystem readings
+	cgroupv2 *Cgroup // used for cgroup prog attachment (and readings, if v2 is default)
+	cgroup   *Cgroup // default cgroup (might point to one or another)
+}
+
+// NewCgroups tries to initialize both: cgroupv1 and cgroupv2, setting one as
+// the default depending on the running environment
+func NewCgroups() (*Cgroups, error) {
+	cs := &Cgroups{}
+
+	isCgroupV1, err := IsCgroupV1Default() // initialize cgroupv1
+	if err != nil {
+		return cs, err
```

Comment:
considering we are returning a pointer, best to return nil if there is an error, no?
if you agree, it would make sense to change all returns of errors on this method

```suggestion
		return nil, err
```

Reply:
We tend to return an empty object in some other places... and I had the same doubt when I did it. Thinking a bit more, after your comment, since I'm setting t.cgroups directly, maybe it is better to have it as nil if not initialized correctly.